autodoc_traits.py

```diff
@@ -177,22 +173,19 @@ def add_directive_header(self, sig):
 
         super().add_directive_header(sig)
 
-    def get_doc(self):
-        """
-        get_doc (get docstring) is called by add_content, which is called by
-        generate. We override it to not unconditionally provide the docstring of
-        the traitlets type, but instead provide the traits help text if its
-        available.
 
-        Links to relevant source code in sphinx.ext.autodoc:
-        - Documenter.generate:             https://github.com/sphinx-doc/sphinx/blob/v6.0.0b2/sphinx/ext/autodoc/__init__.py#L918-L929
-        - AttributeDocumenter.add_content: https://github.com/sphinx-doc/sphinx/blob/v6.0.0b2/sphinx/ext/autodoc/__init__.py#L2655-L2663
-        - Documenter.add_content:          https://github.com/sphinx-doc/sphinx/blob/v6.0.0b2/sphinx/ext/autodoc/__init__.py#L568-L605
-        - AttributeDocumenter.get_doc:     https://github.com/sphinx-doc/sphinx/blob/v6.0.0b2/sphinx/ext/autodoc/__init__.py#L2639-L2653
-        """
-        if isinstance(self.object.help, str):
-            return [[self.object.help]]
-        return super().get_doc()
+def hastraits_attrgetter(obj, name, *defargs):
+    """getattr for trait
+
+    Ensures when HasTraits are documented, their __doc__ attr is defined
+    as the .help string.
+    """
+    attr = safe_getattr(obj, name, *defargs)
+    if isinstance(attr, TraitType):
+        # ensure __doc__ is defined as the trait's help string
+        # if help is empty, that's the same as undocumented
+        attr.__doc__ = attr.help or "No help string is provided."
```

Comment:
Oops, looks like the truthy string is here. I think the intention was to leave this out so undocumented traits would not be included?